include/kllvm/ast/AST.h

```diff
@@ -914,6 +914,9 @@ class kore_definition {
 
   kore_symbol *inj_symbol_{};
 
+  SubsortMap *subsorts_ = nullptr;
+  SubsortMap *subsorts_inverted_ = nullptr;
```

Comment:
Storing raw pointers here is error-prone; as currently written the class will leak the memory allocated to store these maps when it's destroyed. `std::optional` gives you the same interface but correct RAII behaviour.